lib/max_subarray.py

```diff
@@ -2,11 +2,24 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: On
+        Space Complexity: O1
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+    
+    max_so_far = nums[0]
+    max_ending_here = nums[0]
+
+    for i in range(1, len(nums)):
+        max_ending_here = max_ending_here + nums[i]
+        if max_ending_here < 0:
+            max_ending_here = max(max_ending_here, nums[i])
+        if max_so_far < max_ending_here:
+            max_so_far = max_ending_here
```

Comment:
This could be simplified a bit using the `max` function so you wouldn't need the if statements.